homeassistant/components/mqtt/models.py

```diff
@@ -7,19 +7,30 @@
 import attr
 
 PublishPayloadType = Union[str, bytes, int, float, None]
+ReceivePayloadType = Union[str, bytes]
 
 
 @attr.s(slots=True, frozen=True)
-class Message:
+class PublishMessage:
     """MQTT Message."""
 
     topic: str = attr.ib()
     payload: PublishPayloadType = attr.ib()
     qos: int = attr.ib()
     retain: bool = attr.ib()
-    subscribed_topic: str | None = attr.ib(default=None)
-    timestamp: dt.datetime | None = attr.ib(default=None)
 
 
-AsyncMessageCallbackType = Callable[[Message], Awaitable[None]]
-MessageCallbackType = Callable[[Message], None]
+@attr.s(slots=True, frozen=True)
+class ReceiveMessage:
```

Comment:
I'm not too sure about the name, maybe `RxMessage` or `ReceivedMessage` is better?

Reply:
I think it is fine. `Received` could be a tiny more clear, but... it doesn't matter that much to be honest.